cns/restserver/api.go

```diff
@@ -1188,11 +1190,20 @@ func (service *HTTPRestService) publishNetworkContainer(w http.ResponseWriter, r
 		if isNetworkJoined {
 			// Publish Network Container
 
-			pncr := req.CreateNetworkContainerRequestBody
-			pncr.AuthenticationToken = ncParameters.AuthToken
-			pncr.PrimaryAddress = ncParameters.AssociatedInterfaceID
+			innerReqBytes := req.CreateNetworkContainerRequestBody
 
-			err = service.nma.PutNetworkContainer(ctx, &pncr)
+			var innerReq nmagent.PutNetworkContainerRequest
+			err = json.Unmarshal(innerReqBytes, &innerReq)
+			if err != nil {
+				returnMessage = fmt.Sprintf("Failed to publish Network Container: %s", req.NetworkContainerID)
+				returnCode = types.NetworkContainerPublishFailed
+				logger.Errorf("[Azure-CNS] %s", returnMessage)
+			}
```

Comment:
Is this the previous version of the code verbatim? It seems odd that there is no return after this error (the below might blow up). Since it seems like there's another change to be made still to pass linting, would a `return` be appropriate here? If there could be unintended consequences and this is the code as it always had been, then disregard.

Reply:
Fixed in [b866e39](https://github.com/Azure/azure-container-networking/pull/1695/commits/b866e39136e21be893df5c57bb8127a69ae7200a)